### ISSUE TITLE
[flutter_form_builder] When FormBuilder onChanged is called, instantValue is outdated

### ISSUE THREAD

**Issue by mdeandrea-mrmilu** [danvick/flutter_form_builder#1018]
This issue maybe will be from Flutter or only an expected behaviour, but I think that is a necessary discussion.

Expected behaviour:
When read instantValue into method FormBuilder.onChanged, instantValue will be updated

Actual behaviour: 
When read instantValue into method FormBuilder.onChanged, instantValue is outdated (previous value)

Steps to reproduce:

1. Change value in a FormField (ex: FormBuilderTextField)
2. Read instantValue from key (ex: _formKey.currentState?.instantValue)
3. Error: The instantValue is outdated

flutter_form_builder version: `7.1.1`
Device: MacOS - Chrome

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter_form_builder/flutter_form_builder.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  static const String _title = 'Form code sample';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: _title,
      home: Scaffold(
        appBar: AppBar(title: const Text(_title)),
        body: const MyStatefulWidget(),
      ),
    );
  }
}

class MyStatefulWidget extends StatefulWidget {
  const MyStatefulWidget({Key? key}) : super(key: key);

  @override
  State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
}

class _MyStatefulWidgetState extends State<MyStatefulWidget> {
  final _formBuilderKey = GlobalKey<FormBuilderState>();

  @override
  Widget build(BuildContext context) {
    return FormBuilder(
      key: _formBuilderKey,
      onChanged: () {
        print(
            'FormBuilder changed/Instant value: ${_formBuilderKey.currentState?.instantValue}');
      },
      child: FormBuilderTextField(
        name: 'text',
        decoration: const InputDecoration(labelText: 'Text'),
        onChanged: (String? value) => print('FormBuilderText changed: $value'),
      ),
    );
  }
}

```

</details>

Thinks to considered:

- A example use case of this expected behaviour, can be disabled a submit button until has changed
- By definition, onChanged method on Form widget (from Flutter) say this:

> Called when one of the form fields changes.
> In addition to invoking this callback, all the fields of the form will be rebuilt.

**Comment by danvick** [comment]
Hi @mdeandrea-mrmilu,

Please confirm that this issue is fixed in v7.2.1

**Comment by CaLouro** [comment]
I was having the same problem until I updated to 7.2.1. Thanks for it.

This issue might be worth closing since the problem it describes is already solved, although not backwards compatible.
